network/rtcp_packet.ksy

```diff
@@ -0,0 +1,287 @@
+meta:
+  id: rtcp_packet
+  title: rtcp network packet
+  license: CC0-1.0
+  ks-version: 0.7
+
+seq:
+  - id: rtcp_packets
+    type: rtcp_packet
+    repeat: eos
+
+types:
+  rtcp_packet:
+    meta:
+      endian: be
+    seq:
+      - id: b1
+        type: u1
+      - id: payload_type
+        type: u1
+        enum: payload_type
+      - id: length
+        type: u2
+      - id: body
+        size: 4 * length
+        type:
+          switch-on: payload_type
+          cases:
+            'payload_type::sr': sr_packet
+            'payload_type::rr': rr_packet
+            'payload_type::sdes': sdes_packet
+            'payload_type::psfb': psfb_packet
+            'payload_type::rtpfb': rtpfb_packet
+    instances:
+      version:
+        value: (b1 & 0xc0) >> 6
+      padding:
+        value: (b1 & 0x20) >> 5
+      subtype:
+        value: b1 & 0x1f
+
+  sr_packet:
+    meta:
+      endian: be
+    seq:
+      - id: ssrc
+        type: u4
+      - id: ntp_msw
+        type: u4
+      - id: ntp_lsw
+        type: u4
+      - id: rtp_timestamp
+        type: u4
+      - id: sender_packet_count
+        type: u4
+      - id: sender_octet_count
+        type: u4
+      - id: report_block
+        type: report_block
+        repeat: expr
+        repeat-expr: _parent.subtype
+
+  rr_packet:
+    meta:
+      endian: be
+    seq:
+      - id: ssrc
+        type: u4
+      - id: report_block
+        type: report_block
+        repeat: expr
+        repeat-expr: _parent.subtype
+
+  report_block:
+    meta:
+      endian: be
+    seq:
+      - id: ssrc_source
+        type: u4
+      - id: lost_val
+        type: u1
+      - id: highest_seq_num_received
+        type: u4
+      - id: interarrival_jitter
+        type: u4
+      - id: lsr
+        type: u4
+      - id: dlsr
+        type: u4
+    instances:
+      fraction_lost:
+        value: lost_val >> 24
+      cumulative_packets_lost:
+        value: lost_val & 0x00ffffff
+
+  sdes_packet:
+    meta:
+      endian: be
+    seq:
+      - id: source_chunk
+        type: source_chunk
+        repeat: expr
+        repeat-expr: sc_count
+    instances:
+      sc_count:
```

Comment:
The RFC:

>    source count (SC): 5 bits
>       The number of SSRC/CSRC chunks contained in this SDES packet.  A
>       value of zero is valid but useless.


Did 's/sc_count/source_count/g'

Reply:
Well, the general guidance in the style guide is to keep these things consistent across all .ksy files in our repo to make it easier to understand for people without reading the official specs. If you want to convey original IDs, typically that's what `-orig-id` is for — it is totally ignored by the compiler, but you can it can appear in the generated documentation and it would be still available for search in the spec file. Moreover, there are no rules at all checked for `-orig-id` — it can contain spaces, lower/upper case chars, etc. You can do:

```yaml
seq:
  - id: num_source_chunks
    -orig-id: source count (SC)
    type: b5
```

Reply:
Note that `source_count` is a derived field. 
```
  sdes_packet:
    seq:
      - id: source_chunk
        type: source_chunk
        repeat: expr
        repeat-expr: source_count
    instances:
      source_count:
        value: _parent.subtype
```

Adding `orig-id` to an `instance` field craps out: 

```
$ ksc -t python rtcp_packet.ksy
/types/sdes_packet/instances/num_source_chunks/orig-id: invalid key found in value instance, allowed: doc, doc-ref, enum, if, value
Makefile:5: recipe for target 'rtcp_packet.py' failed
make: *** [rtcp_packet.py] Error 2
```

Reply:
It's `-orig-id`, not `orig-id`.